tests/data_context/store/test_profiler_store.py

```diff
@@ -0,0 +1,162 @@
+from great_expectations.data_context.types.resource_identifiers import (
+    ConfigurationIdentifier,
+)
+from great_expectations.rule_based_profiler.config import (
+    DomainBuilderConfig,
+    ExpectationConfigurationBuilderConfig,
+    ParameterBuilderConfig,
+    RuleBasedProfilerConfig,
+    RuleConfig,
+)
+from great_expectations.util import gen_directory_tree_str
+from tests.test_utils import build_profiler_store_using_filesystem
+
+
+@pytest.fixture(scope="function")
+def profiler_name() -> str:
+    return "my_first_profiler"
+
+
+@pytest.fixture(scope="function")
+def store_name() -> str:
+    return "profiler_store"
+
+
+@pytest.fixture(scope="function")
+def rules() -> Dict[str, RuleConfig]:
+    return {
+        "rule_1": RuleConfig(
+            name="rule_1",
+            domain_builder=DomainBuilderConfig(class_name="DomainBuilder"),
+            parameter_builders=[
+                ParameterBuilderConfig(
+                    class_name="ParameterBuilder", name="my_parameter"
+                )
+            ],
+            expectation_configuration_builders=[
+                ExpectationConfigurationBuilderConfig(
+                    class_name="ExpectationConfigurationBuilder",
+                    expectation_type="expect_column_pair_values_A_to_be_greater_than_B",
+                )
+            ],
+        )
+    }
+
+
+@pytest.fixture(scope="function")
+def profiler_config(
+    profiler_name: str, rules: Dict[str, RuleConfig]
+) -> RuleBasedProfilerConfig:
+    return RuleBasedProfilerConfig(name=profiler_name, config_version=1.0, rules=rules)
+
+
+@pytest.fixture(scope="function")
+def empty_profiler_store(store_name: str) -> ProfilerStore:
+    return ProfilerStore(store_name)
+
+
+@pytest.fixture(scope="function")
+def profiler_key(profiler_name: str) -> ConfigurationIdentifier:
+    return ConfigurationIdentifier(configuration_key=profiler_name)
+
+
+@pytest.fixture(scope="function")
+def populated_profiler_store(
+    empty_profiler_store: ProfilerStore,
+    profiler_config: RuleBasedProfilerConfig,
+    profiler_key: ConfigurationIdentifier,
+) -> ProfilerStore:
+    profiler_store = empty_profiler_store
+    profiler_store.set(key=profiler_key, value=profiler_config)
+    return profiler_store
+
+
+def test_profiler_store_raises_error_with_invalid_value(
+    empty_profiler_store: ProfilerStore,
+):
+    with pytest.raises(TypeError):
+        empty_profiler_store.set(
+            key="my_first_profiler", value="this is not a profiler"
+        )
+
+
+def test_profiler_store_set_adds_valid_key(
+    empty_profiler_store: ProfilerStore,
+    profiler_config: RuleBasedProfilerConfig,
+    profiler_key: ConfigurationIdentifier,
+):
+    assert len(empty_profiler_store.list_keys()) == 0
+    empty_profiler_store.set(key=profiler_key, value=profiler_config)
+    assert len(empty_profiler_store.list_keys()) == 1
+
+
+def test_profiler_store_integration(
```

Comment:
where does the cleanup happen?

Reply:
This test relies on `empty_data_context`, which is a function-scoped fixture that cleans up it's `tmp_path` after the individual test is run. The teardown functionality is innate to anything that uses `tmp_path`(https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture)